src/main/daml/Daml/Finance/Interface/Types/Date/Schedule.daml

```diff
@@ -21,8 +21,8 @@ data StubPeriodTypeEnum
 -- | Frequency of a periodic schedule.
 data Frequency = Frequency
   with
-    period : PeriodEnum
-      -- ^ The period, e.g., day, month, etc.
+    period : ScheduleFrequencyEnum
+      -- ^ The period, e.g., day, month, etc (or a single period).
     periodMultiplier : Int
```

Comment:
I would remove the period multiplier and embed it into the `RegularPeriod`. This is because the `periodMultiplier` doesn't make sense for `SinglePeriod`.

Reply:
I started doing this, but it is a rather large change and the resulting data type does not match the FpML one, where period and periodMultiplier are separate. 

I wonder if it is worth doing. Maybe merge what we have at the moment and consider this change separate?

Reply:
I think it is worth doing, otherwise we would be populating an `Int` field with a meaningless value in the single-period case, but happy to discuss with the team.

If the type doesn't match the Fpml one, then I would create a separate type in the Fpml module. `Interface.Types.Date` is a generic module and the data model should not be constrained by the Fpml data model.

Let me know if you prefer to consider the change separately from this PR and I will 🟢

Reply:
I was actually thinking the same thing. Maybe create a `PeriodEnumExtended` FpML type, which contains D, W, M, Y, T (flattened) and then convert that to the new improved Daml Finance types in the FpML util functions. 

That way, specifying an FpML in Daml is very similar to the FpML XML and the types in our library are consistent with our logic.

Reply:
I added a new PeriodExtendedEnum now, corresponding to the FpML Enum with the same name:
https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-shared-5-11_xsd/complexTypes/Frequency/period.html

I decided not to flatten it. This makes it more similar to the FpML one, and it also simplifies translating from the FpML type to the native Daml Finance type.